File: BahamasEngine/BahamasEngine/ExecutionManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BahamasEngine
{
    public abstract class ExecutionManager
    {
        internal Queue<TradingEvent> eventsQueue;
        internal InstrumentDataManager dataManager;

        public ExecutionManager(Queue<TradingEvent> eventsQueue, InstrumentDataManager dataManager)
        {
            this.eventsQueue = eventsQueue;
            this.dataManager = dataManager;
        }

        public virtual void ExecuteOrder(OrderEvent orderEvent) { }
    }

    public class SimulatedExecutionManager : ExecutionManager
    {
        public SimulatedExecutionManager(Queue<TradingEvent> eventsQueue, InstrumentDataManager dataManager)
            : base(eventsQueue, dataManager)
        {
        }

        public override void ExecuteOrder(OrderEvent orderEvent)
        {
            double commission = GetCommission(orderEvent);

            FillEvent tempEvent = new FillEvent(
                orderEvent.Ticker,
                orderEvent.Action,
                orderEvent.OrderUnits,
                dataManager.GetCurrentPrice(orderEvent.Ticker),
                commission,
                orderEvent.PortfolioId);

            eventsQueue.Enqueue(tempEvent);
        }

        private double GetCommission(OrderEvent orderEvent)
        {
            return orderEvent.OrderUnits * 2.35;
        }
    }

}

File: BahamasEngine/execution_manager.py
import queue as queue

import settings
import trading_event as trading_events

class ExecutionManager(object):

    def __init__(self, events_queue, data_manager):
        self.events_queue = events_queue
        self.data_manager = data_manager

    def execute_order(self, o_event):

        f_event = trading_events.FillEvent(
            o_event.option_id,
            o_event.action,
            o_event.units,
            self.data_manager.get_current_price(o_event.option_id),
            settings.commission * o_event.units
        )

        self.events_queue.put(f_event)
